src/lib/table/table.md

```diff
@@ -1,29 +1,42 @@
 The `mat-table` provides a Material Design styled data-table that can be used to display rows of 
 data. 
 
-The table's template consists of two parts: column definitions and row definitions. Each column 
-definition contains templates for that column's header and content cells. Each row definition 
-captures the columns used for that row and any bindings applied to the row element.
-
-A `DataSource` provides data to the table by emitting an `Observable` stream of the items to be 
-rendered. Each emit includes the _entire set of items_ that should be displayed. The table, 
-listening to this stream, will render a row per item. Any manipulation of the data being displayed 
-(e.g. sorting, pagination, filtering) should be captured by the `DataSource`, ultimately emitting 
-a new set of items to reflect any changes.
-
 This table builds on the foundation of the CDK data-table and uses a similar interface for its 
-data source input and template, except that its element and attribute selectors will be prefixed 
-with `mat-` instead of `cdk-`. For detailed information on the interface and how it works, see the 
+data input and template, except that its element and attribute selectors will be prefixed 
+with `mat-` instead of `cdk-`. For more information on the interface and a detailed look at how
+the table is implemented, see the 
 [guide covering the CDK data-table](https://material.angular.io/guide/cdk-table).
 
 ### Getting Started
 
 <!-- example(table-basic) -->
 
-#### 1. Define the table's columns
+#### 1. Write your mat-table and provide data
+
+Begin by creating a `<mat-table>` component in your template and passing in data.
+  
+The simplest way to provide data to the table is by passing a data array to the table's `data` 
+input. The table will take the array and render a row for each object in the data array.
+
+```html
+<mat-table [dataSource]=”myDataArray”>
+  ...
+</mat-table>
+```
+
+Since the table optimizes for performance, it will not automatically check for changes to the data
+array. Instead, when objects are added, removed, or moved on the data array, you can trigger an 
+update to the table's rendered rows by calling its `renderRows()` function. 
+
+See the section "Alternative ways to provide data" for more approaches to providing data to the 
+table.
 
-Start by writing your table's column definitions. Each column definition should be given a unique 
-name and contain the content for its header and row cells.
+#### 2. Define the table's columns
```

Comment:
Would it make sense to say "column templates" (and "row templates" below)? Having "Define the table's rows" feels weird since the data drives the rows, you're just specifying the bindings.